php/proxy.php

```diff
@@ -3,150 +3,192 @@
         }
         return $acrolinxSettings;
     }
-    
+
     private function getPosition($requestURI, $part) {
         return strpos($requestURI, $part);
     }
-    
+
     private function getActualURI($requestURI, $pos, $part) {
         return substr_replace($requestURI, '', 0, ($pos+strlen($part)));
     }
-    
+
+    private function emulate_getallheaders() {
+        foreach ($_SERVER as $name => $value) {
+            if (substr($name, 0, 5) == 'HTTP_') {
+                $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
+                $headers[$name] = $value;
+            } else if ($name == "CONTENT_TYPE") {
+                $headers["Content-Type"] = $value;
+            } else if ($name == "CONTENT_LENGTH") {
+                $headers["Content-Length"] = $value;
+            }
+        }
+        return $headers;
+    }
+
     private function getHeadersOfRequest() {
         $headers = array();
-        $isAuthToken = false; 
-        foreach (getallheaders() as $name => $value) {
-          $headerString = $name.':'.$value;
-
-          //Setting target Server as Host
-          if(strtolower($name) == 'host') {
-              $headerString = $name.':'.parse_url(self::getSettings()['url'])['host'];
-          }
-
-          if($name == 'authToken') {
-            $isAuthToken = true;
-          }
-          array_push($headers,"$headerString");
+        $isAuthToken = false;
+
+        foreach (self::emulate_getallheaders() as $name => $value) {
+
+            $headerString = $name.':'.$value;
+
+            //Setting target Server as Host
+            if(strtolower($name) == 'host') {
+                $headerString = $name.':'.$this->host;
+            }
+            if($name == 'authToken') {
+                $isAuthToken = true;
+            }
+            array_push($headers,"$headerString");
         }
-        
+
         if($isAuthToken == false){
-            $settings = self::getSettings();
-            $username = 'username:'.$settings['username'];
-            $password = 'password:'.$settings['password'];
-            array_push($headers, $username);
-            array_push($headers, $password);
-            array_push($headers,'User-Agent:Tinymce Proxy');
+            $this->settings = self::getSettings();
+            $this->username = 'username:'.$this->settings['username'];
+            $this->password = 'password:'.$this->settings['password'];
+            array_push($headers, $this->username);
+            array_push($headers, $this->password);
+            array_push($headers,'User-Agent:Drupal Proxy');
+
         }
         return $headers;
     }
-    
+
     private function getPOSTdata(){
         return file_get_contents("php://input");
     }
-    
-    private function proxyRequest($url){
+
+    public function __construct() {
+        self::disable_gzip();
```

Comment:
Is that a good idea? Might have a huge performance impact, especially when loading the sidebar through the proxy?

Reply:
Sometimes I have seen that response gets truncated when content is gzipped. This is not reproducible always.
Fix: I have commented this line of code,
//If response is getting truncated try by disabling gzip
//self::disable_gzip();